### PR TITLE
Add mudkip.dev

### PR DESCRIPTION
**Pull request by mudkipdev**
This PR is:

- [X] Adding a new domain
- [ ] Updating existing domain **size**
- [ ] Changing domain name
- [ ] Removing existing domain from list
- [ ] Website code changes (512kb.club site)
- [ ] Other not listed

- [X] I used the uncompressed size of the site
- [X] I have included a link to the Cloudflare report
- [X] This site is not an ultra minimal site
- [X] The following information is filled identical to the data file

***I confirm that I have read the [FAQ section](https://512kb.club/faq), particularly the two red items around minimal pages and inappropriate content and I attest that this site is neither of these things.***

- [X] Check to confirm

```
- domain: mudkip.dev
  url: https://mudkip.dev
  size: 27.33
  last_checked: 2024-05-27
```

Cloudflare Report: https://radar.cloudflare.com/scan/1b7a3bed-f78e-4882-af23-c495bac1886b/summary

### CODE REVIEW

File: _data/sites.yml [hunk 1]
```diff
@@ -2228,7 +2228,7 @@
   size: 26.38
   last_checked: 2024-02-26
 
-- domain: matt-smiley
+- domain: matt-smiley.com
```
Comment (garritfra) [inline _data/sites.yml:2231]: Was this by mistake?

**Review by garritfra — CHANGES_REQUESTED** [review]

File: _data/sites.yml (hunk 1 above)
Comment (mudkipdev) [reply _data/sites.yml:2231]: No, I noticed it wasn't following convention with the other websites. You can remove it if needed

Comment (garritfra) [reply _data/sites.yml:2231]: In which way doesn't it follow the convention? `matt-smiley` is not a valid domain. 😅

**Review by garritfra — APPROVED** [review]

**Comment by garritfra** [comment]
Thanks!

**Pull request merged**